service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -312,6 +312,36 @@ public void deleteGcpCloudContext(UUID workspaceId, AuthenticatedUserRequest use
         .submitAndWait(null);
   }
 
+  /**
+   * Retrieve the optional GCP cloud context
+   *
+   * @param workspaceId workspace identifier of the cloud context
+   * @return optional GCP cloud context
+   */
+  public Optional<GcpCloudContext> getGcpCloudContext(UUID workspaceId) {
```

Comment:
aside: I'm wondering if we've missed an opportunity for an abstraction to keep the functions that work on a particular cloud platform together. It feels like a really fundamental categorization that's buried in the `Gcp` infix. Maybe there could be separate interfaces for the two clouds, or even separate services. In other words, representing/enforcing a major abstraction with a naming convention only seems like a code smell. Just thinking out loud.

Reply:
Worth thinking further about